woodwork/schema.py

```diff
@@ -426,13 +438,16 @@ def _filter_cols(self, include, col_names=False):
                 tags_used.add(selector)
             # Determine if the selector is a column name
             if col_names and selector in self.columns:
-                cols_to_include.add(selector)
+                match_cols.add(selector)
 
         for col_name, col in self.columns.items():
             if _get_ltype_class(col['logical_type']) in ltypes_used or col['semantic_tags'].intersection(tags_used):
-                cols_to_include.add(col_name)
+                match_cols.add(col_name)
```

Comment:
What would you think of sticking with a `cols_to_include` list of sorts (though maybe need a better name) and just determine what cols get added to it by whether or not we're excluding or including:

```python
for col_name, col in self.columns.items():
    is_match = get_ltype_class(col['logical_type']) in ltypes_used or col['semantic_tags'].intersection(tags_used)
    if include and is_match:
         cols_to_include.add(col_name)
    elif exclude and not is_match:
         cols_to_include.add(col_name)         
```

Really the only benefit of this way is that it lets us avoid iterating over the columns again for `exclude` below (which is only a problem if there's a lot of columns), so I don't know that there's a huge benefit to this way, but I thought I'd bring it up!

Reply:
Good thought. Had to make some other updates to account for the logic around column name matches, but should now  iterate through the columns only once.